src/BigDecimal.ts

```diff
@@ -0,0 +1,717 @@
+ * assert.deepStrictEqual(parse('123.abc'), none())
+ *
+ * @since 2.0.0
+ * @category constructors
+ */
+export const parse = (s: string): Option.Option<BigDecimal> => {
+  let digits: string
+  let scale: number
+
+  const dot = s.search(/\./)
+  if (dot !== -1) {
+    const lead = s.slice(0, dot)
+    const trail = s.slice(dot + 1)
+    digits = `${lead}${trail}`
+    scale = trail.length
+  } else {
+    digits = s
+    scale = 0
+  }
+
+  if (digits === "") {
+    return Option.some(zero)
+  }
+
+  if (!/^(?:\+|-)?\d+$/.test(digits)) {
+    return Option.none()
+  }
+
+  return Option.some(scaled(BigInt(digits), scale))
+}
+
+/**
+ * Normalizes a given `BigDecimal` by removing trailing zeros.
+ *
+ * @param self - The `BigDecimal` to normalize.
+ *
+ * @example
+ * import { normalize, make } from 'effect/BigDecimal'
+ *
+ * assert.deepStrictEqual(normalize(make(123.456)), make(123.456))
+ * assert.deepStrictEqual(normalize(make(123.456000)), make(123.456))
+ * assert.deepStrictEqual(normalize(make(123.000456)), make(123.000456))
+ * assert.deepStrictEqual(normalize(make(123.000000)), make(123))
+ *
+ * @since 2.0.0
+ * @category constructors
+ */
+export const normalize = (self: BigDecimal): BigDecimal => {
+  if (self.value === 0n) {
+    return zero
+  }
+
+  const digits = `${self.value}`.split("")
+
+  let trail = 0
+  for (let i = digits.length - 1; i >= 0; i--) {
+    if (digits[i] === "0") {
+      trail++
+    } else {
+      break
+    }
+  }
+
+  if (trail === 0) {
+    return self
+  }
+
+  digits.splice(digits.length - trail)
+  const value = BigInt(digits.join(""))
+  const scale = self.scale - trail
+
+  return scaled(value, scale)
+}
+
+/**
+ * Scales a given `BigDecimal` to the specified scale.
+ *
+ * If the given scale is smaller than the current scale, the value will be rounded down to
+ * the nearest integer.
+ *
+ * @param self - The `BigDecimal` to scale.
+ * @param scale - The scale to scale to.
+ *
+ * @since 2.0.0
+ * @category constructors
+ */
+export const scale = (self: BigDecimal, scale: number): BigDecimal => {
+  if (scale > self.scale) {
+    return scaled(self.value * 10n ** BigInt(scale - self.scale), scale)
+  }
+
+  if (scale < self.scale) {
+    return scaled(self.value / 10n ** BigInt(self.scale - scale), scale)
+  }
+
+  return self
+}
+
+/**
+ * Provides an addition operation on `BigDecimal`s.
+ *
+ * @param self - The first operand.
+ * @param that - The second operand.
+ *
+ * @example
+ * import { sum, make } from 'effect/BigDecimal'
+ *
+ * assert.deepStrictEqual(sum(make(2n), make(3n)), make(5n))
+ *
+ * @since 2.0.0
+ * @category math
+ */
+export const sum: {
+  (that: BigDecimal): (self: BigDecimal) => BigDecimal
+  (self: BigDecimal, that: BigDecimal): BigDecimal
+} = dual(2, (self: BigDecimal, that: BigDecimal): BigDecimal => {
+  if (that.value === 0n) {
+    return self
+  }
+
+  if (self.value === 0n) {
+    return that
+  }
+
+  if (self.scale > that.scale) {
```

Comment:
Getting equivalent scales is quite common, so maybe a helper could be added?

```ts
const [a, b] = whateverify(self, that)
```

Reply:
I thought about that but since these are operations that might get called countless times I also wanted to keep things as optimized as possible. Allocating a new array instance (sometimes even needlessly if the scale is already identical) seemed wasteful. Microoptimization, sure, but it is at the data type level after all.

Reply:
👍🏼